src/decrypt-api/Startup.cs

```diff
@@ -180,5 +180,16 @@ private IKeyManagement GetGoogleCloudKeyManagment()
                 location,
                 protectionLevel);
         }
+
+        private IKeyManagement GetAwsKeyManagement()
+        {
+            var awsKey = Configuration.GetValue<string>("KeyManagement:AwsKms:Key");

```

Comment:
What are the authentication options? We need to support also authentication with instance metadata instead of supplying credentials. Usually, AWS SDK provides some magic that knows to create authentication based on the environment (e.g. files on the disk, env var etc) and choose the most relevant one.

Reply:
Authentication options are pretty limited inside a pod.
There are some tools to map IAM role to a pod, but I don't think we need to support that on the first implementation.

Reply:
Safely assume that a pod has access to AWS metadata service (169.254.169.254) with a credentials on it. A credentials supply chain in a typical AWS client should have metadata service src in it.
https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/net-dg-config-creds.html
see `8-EC2 instance metadata.`. Note that it's the last in the chain, so you can still override it with a file, env and so on. This should be handled entirely by an SDK.

An IAM-role-to-pod mapper (like https://github.com/uswitch/kiam) is an absolute must in every AWS cluster, I think everyone have it already.

Reply:
thanks for your help @max-lobur ! 
I'll make the default behavior to let the SDK handle the authentication, and let you override it with user access key and secret for users that running their cluster outside AWS and want to use AWS KMS.